src/doc/rustdoc/src/linking-to-items-by-name.md

```diff
@@ -76,19 +96,66 @@ struct Foo {}
 fn Foo() {}
 ```
 
+The following prefixes can be used:
+
+- `struct@`
+- `enum@`
+- `type@`
+- `trait@`
+- `union@`
+- `const@`
+- `static@`
+- `value@`
+- `fn@` / `function@` / `method@`
+- `mod@` / `module@`
+- `prim@` / `primitive@`
+- `macro@`
+- `derive@`
+
+These prefixes will be stripped when displayed in the documentation, so `[struct@Foo]`
+will be rendered as `Foo`.
+
 You can also disambiguate for functions by adding `()` after the function name,
 or for macros by adding `!` after the macro name:
 
 ```rust
-/// See also: [`Foo`](struct@Foo)
-struct Bar;
+/// This is different from [`foo!`]
+fn foo() {}
 
-/// This is different from [`Foo()`]
-struct Foo {}
+/// This is different from [`foo()`]
+macro_rules! foo {
+  () => {}
+}
+```
 
-fn Foo() {}
+## Warnings, re-exports, and scoping
+
+Links are resolved in the scope of the module where the item is defined, even
+when the item is re-exported. If a link from another crate fails to resolve, no
+warning is given.
+
+When re-exporting an item, rustdoc allows adding additional documentation to it.
+That additional documentation will be resolved in scope of the re-export, not
+the original, allowing you to link to items in the new crate. The new links
+will still give a warning if they fail to resolve.
```

Comment:
These two paragraph seem contradictory. The first one says that links are resolved in the module where the item is defined, even when it's re-exported, while the second says that documentation you add on a re-export will be resolved in the re-export crate. Which one is it? :P

Reply:
The *original* links will be resolved in the original scope. The *new* links will be resolved in the new scope. The idea is for this to work:

```rust
// crate 1
/// Link to [f()]
pub struct S;
pub fn f() {}

// crate2
/// Link to [g()]
pub use crate1::S;
pub fn g() {}

Reply:
I think you should say something like that in the docs, because I couldn't tell what you meant from the way it is right now.

Reply:
I couldn't think of a way to explain this better, but I did add an example of both scenarios.